Library/Homebrew/extend/os/mac/formula_cellar_checks.rb

```diff
@@ -58,36 +56,10 @@ def check_python_framework_links(lib)
     EOS
   end
 
-  def check_linkage
-    return unless formula.prefix.directory?
-    keg = Keg.new(formula.prefix)
-    checker = LinkageChecker.new(keg, formula)
-
-    if checker.broken_dylibs?
-      audit_check_output <<-EOS.undent
-        The installation was broken.
-        Broken dylib links found:
-          #{checker.broken_dylibs.to_a * "\n          "}
-      EOS
-    end
-
-    # only check undeclared deps for standard installations.
```

Comment:
I think there has been universal agreement that the broken dylib check is actually helpful (please correct me if I'm wrong), i.e. everything above this line. That's basically `brew linkage --test` for the formula itself that is being linked (if viewed from the perspective of `brew test-bot`).

Everything from this line to the end of the method is the controversial and problematic check for undeclared dependencies and I'd be in favor of only removing this second chunk.